rewrite-yaml/src/test/java/org/openrewrite/yaml/ChangePropertyKeyTest.java

```diff
@@ -778,4 +778,28 @@ void doesNotBreakOnKeysWhichIncludeRegexSpecialCharacters() {
         );
     }
 
+    @Test
+    @Issue("https://github.com/openrewrite/rewrite/issues/2881")
+    void embedIndentedPropertyIntoExisting() {
+        rewriteRun(
+          spec -> spec.recipe(new ChangePropertyKey("a.b.d", "a.b", null, null, null)),
+          yaml(
+            """
+              a:
+                b:
+                  c: false
+                  d:
+                    c: true
```

Comment:
Was it a conscious choice to deviate slightly from the sample given in #2881 to show the merge behavior with clashing fields such as `c: true` here? Like how you leave no doubt about the impact of your implementation in those cases.

Reply:
Ah, yes, I changed the test. Clashes can possibly occur with the current behavior of keeping the "old" values. This could be changed if `acceptTheirs` of the `MergeYamlVisitor` is set to `false`.